src/test.js

```diff
@@ -834,8 +919,10 @@ export function internalStop( test ) {
 						`Test took longer than ${timeout}ms; test timed out.`,
 						sourceFromStacktrace( 2 )
 					);
-					released = true;
-					internalRecover( test );
+
+					pause.killed = true;
+					test.asyncPauses.delete( pauseId );
```

Comment:
Should these lines appear before invoking `pushFailure`? I'm wondering if there's any chance of `pushFailure` prompting code that can observe the not-yet-abandoned async pause.

Reply:
Yep. I'll also make it call `test.pushFailure()` directly here.